n_helpers.c

```diff
@@ -150,60 +287,87 @@ const char *NoteBinaryEncode(const uint8_t *inBuf, uint32_t inLen,
-          For a constant-time alternative, see NoteBinaryEncodedMaxLength.
-  @returns  The required buffer size to hold the encoded data.
-*/
+  @brief  Compute the maximum decoded data length guaranteed
+          to fit into a fixed-size buffer, after being encoded.
+
+          This API is designed for a space constrained environment, where a
+          working buffer has been allocated to facilitate with binary
+          transactions.
+
+          There are two primary use cases:
+
+          1. When data is retrieved from the Notecard, it must be requested in
+             terms of the unencoded offset and length. However, the data is
+             encoded prior to transmission, and, as a result, the buffer must be
+             capable of receiving the encoded (larger) data. This API returns a
+             length that is safe to request from the Notecard, because the
+             resulting encoded data is guaranteed to fit in the provided buffer.
+          2. When data is transmitted to the Notecard, this API can be used to
+             verify whether or not unencoded data of a given length will fit in
+             the provided buffer after encoding.
+
+  @param  size The size of the fixed-size buffer.
+
+  @returns  The max length of unencoded data certain to fit in the fixed-size
+            buffer, after being encoded.
+ */
 /**************************************************************************/
-uint32_t NoteBinaryEncodedLength(const uint8_t *buf, uint32_t len)
+uint32_t NoteBinaryMaxDecodedLength(uint32_t bufferSize)
 {
-    return cobsEncodedLength(buf, len);
+    return cobsGuaranteedFit(bufferSize);
 }
 
 //**************************************************************************/
 /*!
-  @brief  Given an input buffer's length, compute the maximum output buffer size
-          needed to binary encode the input buffer.
-  @param  len The length of the input buffer.
+  @brief  Compute the maximum buffer size needed to encode
+          any unencoded buffer of the given length.
+
+  @param  unencodedLength The length of an unencoded buffer.
+
   @returns  The max required buffer size to hold the encoded data.
-*/
+ */
 /**************************************************************************/
-uint32_t NoteBinaryEncodedMaxLength(uint32_t len)
+uint32_t NoteBinaryMaxEncodedLength(uint32_t unencodedLength)
 {
-    return cobsEncodedMaxLength(len);
+    return cobsEncodedMaxLength(unencodedLength);
 }
 
 //**************************************************************************/
 /*!
   @brief  Receive a large binary object from the Notecard's binary buffer
+
   @param  buffer A buffer to hold the binary object
   @param  bufLen The total length of the provided buffer
-  @param  dataLen An out parameter to hold the length of the decoded data in the
-                  output buffer.
+  @param  decodedOffset The offset to the decoded binary data already residing
```

Comment:
I thought you were gonna go with `notecardOffset`.

Reply:
That was for `NoteBinaryTransmit()`. It is redundant on a receive, and I believe it is more important to understand that data on the Notecard is in a decoded state when trying to calculate offsets and lengths.